File: app/src/main/java/com/example/myapplication/Utils/SharedPreferenceBase.java
package com.example.myapplication.utils;

import android.content.Context;
import android.content.SharedPreferences;

import static android.content.Context.MODE_PRIVATE;

public class SharedPreferenceBase {
    private SharedPreferences pref;
    private SharedPreferences.Editor editor;
    public Context context;
    private static SharedPreferenceBase sharedPreferenceBase = new SharedPreferenceBase();

    public static SharedPreferenceBase getInstance() {
        return sharedPreferenceBase;
    }

    public Context getContext() {
        return context;
    }

    public void setContext(Context context) {
        this.context = context;
        pref = this.context.getSharedPreferences("pref", MODE_PRIVATE);
        editor = pref.edit();
    }

    public void setString(String key, String value) {

        editor.putString(key, value);
        editor.commit();
    }

    public void setBoolean(String key, Boolean value) {
        editor.putBoolean(key, value);
        editor.commit();
    }

    public void setInt(String key, int value) {

        editor.putInt(key, value);
        editor.commit();
    }

    public String getString(String key, String value) {
        pref = context.getSharedPreferences("pref", MODE_PRIVATE);
        return pref.getString(key, value);
    }

    public int getInt(String key, int value) {
        pref = context.getSharedPreferences("pref", MODE_PRIVATE);
        return pref.getInt(key, value);
    }

    public Boolean getBoolean(String key, boolean value) {
        pref = context.getSharedPreferences("pref", MODE_PRIVATE);
        return pref.getBoolean(key, value);
    }

    public void setLong(String key, long value) {
        editor.putLong(key, value);
        editor.commit();
    }

    public long getLong(String key, long value) {
        pref = context.getSharedPreferences("pref", MODE_PRIVATE);
        return pref.getLong(key, value);
    }

}

File: app_kotlin/src/main/java/com/example/myapplication/utils/SharedPreferenceBase.kt
package com.example.myapplication.utils

import android.content.Context
import android.content.SharedPreferences

import android.content.Context.MODE_PRIVATE

class SharedPreferenceBase {
    private var pref: SharedPreferences? = null
    private var editor: SharedPreferences.Editor? = null
    var context: Context? = null
    set(value) {
        field = value
        pref = this.context?.getSharedPreferences("pref", MODE_PRIVATE)
        editor = pref!!.edit()

    }

//    fun getContext(): Context {
//        return context
//    }

//    fun setContext(context: Context) {
//        this.context = context
//        pref = this.context.getSharedPreferences("pref", MODE_PRIVATE)
//        editor = pref!!.edit()
//    }

    fun setString(key: String, value: String) {

        editor!!.putString(key, value)
        editor!!.commit()
    }

    fun setBoolean(key: String, value: Boolean?) {
        editor!!.putBoolean(key, value!!)
        editor!!.commit()
    }

    fun setInt(key: String, value: Int) {

        editor!!.putInt(key, value)
        editor!!.commit()
    }

    fun getString(key: String, value: String): String? {
        pref = context?.getSharedPreferences("pref", MODE_PRIVATE)
        return pref!!.getString(key, value)
    }

    fun getInt(key: String, value: Int): Int {
        pref = context?.getSharedPreferences("pref", MODE_PRIVATE)
        return pref!!.getInt(key, value)
    }

    fun getBoolean(key: String, value: Boolean): Boolean? {
        pref = context?.getSharedPreferences("pref", MODE_PRIVATE)
        return pref!!.getBoolean(key, value)
    }

    fun setLong(key: String, value: Long) {
        editor!!.putLong(key, value)
        editor!!.commit()
    }

    fun getLong(key: String, value: Long): Long {
        pref = context?.getSharedPreferences("pref", MODE_PRIVATE)
        return pref!!.getLong(key, value)
    }

    companion object {
        val instance = SharedPreferenceBase()
    }

}